src/library_syscall.js

```diff
@@ -1830,8 +1827,20 @@ if (SYSCALL_DEBUG) {
 var WASI_SYSCALLS = set([
   'fd_write',
   'fd_close',
+  'fd_read',
 ]);
 
+// Fallback for cases where the wasi_unstable.name prefixing fails,
+// and we have the full name from C. This happens in fastcomp (which
+// lacks the attribute to set the import module and base names) and
+// in LTO mode (as bitcode does not preserve them).
+
+if (!WASM_BACKEND || !WASM_OBJECT_FILES) {
```

Comment:
I guess I have the same question here now as I did on #9555

Reply:
There's a [bug filed](https://bugs.llvm.org/show_bug.cgi?id=43211) and @sbc100 is working on it, I believe, but I don't think we know why it's broken yet.

Reply:
Ah, I think I just missed the bug. Could you put a link to that in the comment?

Reply:
Added links.